### ISSUE TITLE
Support double quotes for strings

### ISSUE THREAD

**Issue by HDembinski** [mesonbuild/meson#11239]
tl;dr: Please support doubles for strings in addition to single quotes. Enforcing single quotes goes against muscle memory that every C/C++ dev has. It creates a stumbling block when a dev switches between the program language and the build language. Meson is all about productivity and this is not productive.

Context:

Hi, I am the author of Boost.Histogram and several scientific Python/C++ libraries, I contributed to scipy and matplotlib.

I tried out meson yesterday for the first time and ported one of my projects from CMake to meson. I want to use meson-python, the PEP-517 compliant build backend for Python packages. Also see my other issue #11238.

Meson is very opinionate about a lot of things like names for directories and other bike-shedding material. It is a bit annoying, but I saw the rationale for some of these choices and I do admit that enforcing some consistency between projects instead of relying on "style" is a good thing.

However, I think it is bad that meson enforces these use of single quotes for strings. I can see no good reason for that.

- As a C/C++ dev, it is simply muscle memory to use double quotes everywhere. Enforcing single quotes in meson.build files creates a stumbling block when a dev switches between the program language and the build language. meson is all about productivity and this is not productive.
- Practially every language in the world either uses double quotes or supports both single and double quotes
- Even Black, the opinionated Python formatter, tried to establish single quotes for Python, but eventually switched to double quotes.

**Comment by createyourpersonalaccount** [comment]
Personally I like fewer options, it provides a sense of security. I will not have to look up the difference between single and double quotes in meson. I will not have to uniformize the quoting style of meson files, or deal with commits from people who did.

Meson strings are simple, and I think the choice to use single quotes conveys this design decision, e.g. no substitution in `'$foo'`.

**Comment by jpakkane** [comment]
The reason Meson only uses single quotes is for simplicity and also _exactly_ because almost every other system uses double quotes by default. The reason for this is that it is a fairly common need to embed double quotes within strings for things like '-Dfoo="bar"`. With single quotes you never have to care about escaping those so that they survive shell invocations and all that. This makes the common case simpler and more readable. The unfortunate downside is that this is a bit inconvenient for some people, but you can't have your cake and eat it too.

Support for double quotes is unlikely to be added unless there is some technical reason, that is, some case where single quotes simply can not express some required feature. This is unlikely to happen.

**Comment by tristan957** [comment]
I think it is a huge mistake to support `'` and `"`. Meson is good because there is generally only way to do something. This is a `-1` for me. If someone doesn't want to use Meson because it doesn't support `"`, that would seem very illogical.

**Comment by HDembinski** [comment]
I think the people here who are against this are writing Python only or some other language that allows single quotes for strings. If you write C/C++, which two extremely common languages, you have to use double quotes. It is great that Python supports both styles of quotes, so that one does not have to switch every time. Using shift + ' is muscle memory.

**Comment by bonzini** [comment]
The "muscle memory" argument is quite weak. Personally I've gotten accustomed to black's double quotes fairly easily and I believe that double quotes do help for cases like `warning()`, on the other hand I would accept if this bug was closed.

But keeping it open is the worst of both worlds, so someone (not me, I'm just a bug monkey) should make a decision.

**Comment by HDembinski** [comment]
I gave up on meson because of how this convo here played out, and I certainly won't recommend it, because of the mindset it reveals. If the devs don't care about UX, then there is no point in adopting meson, whose whole purpose IMHO is to make building projects more enjoyable. UX is the only reason for me to try out meson over cmake, which otherwise has many advantages (wide adoption for one).

**Comment by bonzini** [comment]
I went through a bunch of projects, and all I found is errors and warnings, like:

```
  error('ssh proxy requires vm_sockets.h which wasn\'t found') # libvirt
  error('SPIRV-Tools doesn\'t contain https://github.com/KhronosGroup/SPIRV-Tools/pull/5534\n') # mesa
  error('Expected CONFIG_VAL1 to be set, but it wasn\'t') # meson test suite
```

Not having to quote apostrophes in English text would be nice but not really earth shattering. There wasn't a single use for a -D option. Closing.
